packages/mui-joy/src/Select/Select.tsx

```diff
@@ -571,16 +571,27 @@ const Select = React.forwardRef(function Select<OptionValue extends {}, Multiple
     [listboxProps.modifiers],
   );
 
+  const renderDisplayValue = () => {
```

Comment:
Does this really need to be a function? Can't we just calculate the value and use it directly?

Reply:
since there are nested `if` statements i went function route as it is more readable. if not for `function` code would look like this, 
```

  {Array.isArray(selectedOption)
            ? selectedOption.length > 0
              ? renderValue(selectedOption)
              : placeholder
            : selectedOption
            ? renderValue(selectedOption)
            : placeholder}
```

Also there is eslint rule to disallow nested ternary.

Reply:
Agreed, calculating value is simpler. removed function